File: synchronization/sources/atomic.4.f90
! @@name:	atomic.4
! @@type:	F-free
! @@operation:	compile
! @@expect:	success
! @@version:	omp_5.0
subroutine boxster(box_totals, vals, box, N)
  use omp_lib
  external calc_val
  real,    intent(inout) :: box_totals(:)
  real,    intent(in)    :: vals(:)
  integer, intent(in)    :: box(:)
  integer                :: N, idx

    !$omp parallel do
    do idx=1,N
      call calc_val(vals(idx))
      !$omp atomic hint(omp_sync_hint_uncontended)
      box_totals( box(idx) ) = box_totals( box(idx) ) + vals(idx)
    enddo

end subroutine

File: synchronization/sources/atomic.4.c
/*
* @@name:	atomic.4
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_5.0
*/
#include <omp.h>

void calc_val(float *val);

void boxster(float *box_totals, float *vals, int *box, int N)
{
    #pragma omp parallel for
    for(int idx=0; idx<=N; idx++)
    {
       calc_val(&vals[idx]);
       #pragma omp atomic hint(omp_sync_hint_uncontended)
       box_totals[ box[idx] ] = box_totals[ box[idx] ] + vals[idx];
    }
}
